extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -375,24 +375,15 @@ public WebDeploymentVerticle(int port, int httpsPort, String host, HttpServerOpt
         @Override
         public void start(Future<Void> startFuture) throws Exception {
             final AtomicInteger remainingCount = new AtomicInteger(httpsOptions != null ? 2 : 1);
+            final HttpServerStartHandler httpServerStartHandler = new HttpServerStartHandler(startFuture, httpOptions,
+                    remainingCount);
```

Comment:
I think you went a bit fast here with the refactoring. One deals with `httpOptions`, the other with `httpsOptions`.

Reply:
You're right! Let me fix that

Reply:
Fixed

Reply:
I had to revert the refactoring because there are now changes in the handling of the 2 cases. So grouping them doesn't make a whole lot of sense anymore